### PR TITLE
CLOUDDST-32867: Add build-arg param to fbc-inject-lifecycle

### PR DESCRIPTION
**Pull request by asergienk**
Add an optional BUILD_ARGS param to fbc-inject-lifecycle task in case product teams use ARGs in their Dockerfile that are resolved at build time.

In `inject-lifecycle` had to switch from command/args to script to support BUILD_ARGS. It would've been possible to keep command/args and still add BUILD_ARGS support, but it would've required that extra dummy positional argument to avoid silently dropping the first build-arg — an easy mistake to make and a subtle one to debug if missed.

### Risk Assesement
Low

We are just adding a new optional ARG, there is no logic change of what the task does.

### CODE REVIEW

File: task/fbc-inject-lifecycle-oci-ta/0.1/fbc-inject-lifecycle-oci-ta.yaml [hunk 1]
```diff
@@ -27,6 +27,11 @@ spec:
       description: Path to the Dockerfile to build.
       name: DOCKERFILE
       type: string
+    - name: BUILD_ARGS
+      description: Array of --build-arg values ("arg=value" strings) used to
+        resolve ARG references in the base image tag.
+      type: array
+      default: []
```
[the remark this answers is not in the document]
Comment (asergienk) [reply task/fbc-inject-lifecycle-oci-ta/0.1/fbc-inject-lifecycle-oci-ta.yaml:37]: all the tests are in operator-foundry repo

[the remark this answers is not in the document]
Comment (asergienk) [reply task/fbc-inject-lifecycle-oci-ta/0.1/fbc-inject-lifecycle-oci-ta.yaml:37]: there is currently no non-oci-ta variant

[the remark this answers is not in the document]
Comment (asergienk) [reply task/fbc-inject-lifecycle-oci-ta/0.1/fbc-inject-lifecycle-oci-ta.yaml:37]: fixed

**Comment by asergienk** [comment]
fullsend approved this PR: 🤖 Finished Review · ✅ Success · Started 9:51 PM UTC · Completed 10:03 PM UTC
the CI fails due to unrelated to the PR reason: 
```
• Submitting APPROVE review
    Attaching 1 inline comment(s)
Error: submitting review: create pull request review on #310: github api: 422 Unprocessable Entity
Error: fullsend post-review failed with exit code 1 (PR #310 in konflux-ci/konflux-operator-tasks)
  ✗ Post-script failed: exit status 1
  ! Failed to remove download dir: unlinkat /tmp/agent-review-3557-1784918892/.github/workflows/fullsend.yaml: permission denied
Error: post-script /home/runner/work/konflux-operator-tasks/konflux-operator-tasks/.fullsend/.fullsend-cache/resources/sha256/56c5ec5912aab02372b7d7f76b422bae167639e175977c63cbb5c89fd2a8e332/content failed: exit status 1
Error: Process completed with exit code 1.
```

**Review by yashvardhannanavati — APPROVED** [review]

**Review by chandwanitulsi — APPROVED** [review]

**Comment by asergienk** [comment]
> Findings
> Medium
> [test-adequacy] task/fbc-inject-lifecycle-oci-ta/0.1/fbc-inject-lifecycle-oci-ta.yaml — No ShellSpec tests or Tekton integration tests exist for this task (no spec/ or tests/ directory under task/fbc-inject-lifecycle-oci-ta/0.1/). CLAUDE.md states "Every change to a task must include or update tests." This PR adds non-trivial logic: a BUILD_ARG_FLAGS loop that converts positional arguments into --build-arg flags (repeated across three steps), and changes the inject-lifecycle step from command/args to script form. These changes should be covered by tests to verify correct behavior with empty, single, and multiple build arguments.
> Remediation: Add ShellSpec unit tests covering: (1) empty BUILD_ARGS produces no --build-arg flags, (2) a single BUILD_ARG is correctly passed as --build-arg key=value, (3) multiple BUILD_ARGS each produce their own --build-arg flag, (4) values containing special characters (spaces, equals signs) are correctly quoted.
> Low
> [missing-documentation] task/validate-fbc/0.2/MIGRATION.md — The migration script at task/validate-fbc/0.2/migrations/0.2.sh injects fbc-inject-lifecycle-oci-ta into FBC pipelines but does not pass the BUILD_ARGS parameter. MIGRATION.md does not mention that users may need to manually configure BUILD_ARGS if their Dockerfile uses ARG directives. Since BUILD_ARGS defaults to an empty array, this is not a functional issue — the task works correctly without it — but a note would help users whose Dockerfiles require build arguments.


1. [test-adequacy] - all tests are in the operator-foundry repo. The PRs that add additional test cases for this new flag are:
https://github.com/konflux-ci/operator-foundry/pull/33
https://github.com/konflux-ci/operator-foundry/pull/34

2. [missing-documentation] - addeda Note in MIGRATION.md

File: task/validate-fbc/0.2/MIGRATION.md [hunk 2]
```diff
@@ -24,3 +24,14 @@ No action required. Review and merge the MintMaker PR once CI passes.
 
 If the migration script fails, manually add `fbc-inject-lifecycle` following the
 standard FBC pipeline blueprint in `build-definitions`.
+
```
[the remark this answers is not in the document]
Comment (asergienk) [reply task/validate-fbc/0.2/MIGRATION.md:27]: that's expected, the migration script for `validate-fbc` covers the roll out of `fbc-inject-lifecycle`

**Comment by maneeshmehra** [comment]
Approved.
